src/viking.js

```diff
@@ -1,11 +1,162 @@
+    this.health = health
+    this.strength = strength }
+
+    //metodos viking
+
+    receiveDamage(damage){
+        
+        this.health = this.health - damage
+        let isStillAlive = `${this.name} has received ${damage} points of damage`
+        let vikingDies= `${this.name} has died in act of combat`
+        if ( this.health > 0 ) {
+            return isStillAlive
+        } else {
+            return vikingDies
+        }
+    }
+    
+    battleCry(){
+        return `Odin Owns You All!`
+
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
 
+//metodos Saxon
+
+receiveDamage(damage){
+        
+    this.health = this.health - damage
+    let isStillAlive = `A Saxon has received ${damage} points of damage`
+    let saxonDies= `A Saxon has died in combat`
+    if ( this.health > 0 ) {
+        return isStillAlive
+    } else {
+        return saxonDies
+    }
+}
+}
 // War
-class War {}
+class War {
+    
+    constructor () {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+
+    };
+//metodo agregar un viking al ejercito viking
+addViking (vikingo1) {
+    this.vikingArmy.push (vikingo1)
+    console.log (this.vikingArmy)
+};
+
+
+ 
+//metodo agregar un saxon al ejercito saxon 
+ addSaxon (saxon1) {
+    this.saxonArmy.push(saxon1)
+    console.log (this.saxonArmy)
+ };
+
+
+
+
+
+
+//ataque vikingo
+vikingAttack(){
+    let vikingRamdom = this.vikingArmy[Math.floor(Math.random()*this.vikingArmy.length)];console.log(vikingRamdom)
+    let saxonRamdom =this.saxonArmy[Math.floor(Math.random()*this.saxonArmy.length)];console.log(saxonRamdom)
+    
+
+    let damage=vikingRamdom.strength
+    let indexSaxonRandom= this.saxonArmy.indexOf(saxonRamdom)
+    console.log(indexSaxonRandom) 
+    if (damage >= saxonRamdom.health){ 
+        this.saxonArmy.splice(indexSaxonRandom,1)
+        
+    }
+    console.log(vikingRamdom.strength)
+    console.log(saxon2.health)
+    return saxonRamdom.receiveDamage(damage)
+};
+//ataque saxon
+saxonAttack(){
+    let vikingRamdom = this.vikingArmy [Math.floor (Math.random () *this.vikingArmy.length )] ;console.log(vikingRamdom)
+    let saxonRamdom = this.saxonArmy [Math.floor (Math.random () *this.saxonArmy.length )] ;console.log(saxonRamdom)
+    let damage = saxonRamdom.strength;console.log(damage)
+    let indexVikingRandom = this.vikingArmy.indexOf (vikingRamdom) ;console.log(indexVikingRandom)
+    console.log(vikingRamdom.health)
+    if (damage >= vikingRamdom.health ) {
+        this.vikingArmy.splice( indexVikingRandom,1)
+    }
+
+    
+    return vikingRamdom.receiveDamage(damage) 
+};
+//show status
+showStatus(){
+if (this.vikingArmy.length <= 0) {
+    return `Saxons have fought for their lives and survived another day...`
+}else if (this.saxonArmy.length <= 0){
+    return `Vikings have won the war of the century!`
+}else if (this.vikingArmy.length > 0 && this.saxonArmy.length > 0){
+    return `Vikings and Saxons are still in the thick of battle.`
+}
+
+
+}
+
```

Comment:
Si soy capaz de leerlo salud mental ! :D
```suggestion
showStatus(){
    if (this.vikingArmy.length <= 0) {
        return `Saxons have fought for their lives and survived another day...`
    } else if (this.saxonArmy.length <= 0){
        return `Vikings have won the war of the century!`
    } else if (this.vikingArmy.length > 0 && this.saxonArmy.length > 0){
        return `Vikings and Saxons are still in the thick of battle.`
    }
}
```